### ISSUE TITLE
Incorrect duplicate parameter name error

### ISSUE THREAD

**Issue by pie-flavor** [intellij-rust/intellij-rust#3351]
## Environment

* **Intellij-Rust plugin version:** v0.2.0.2114-183
* **Rust toolchain version:** rustc 1.33.0-nightly (a2b0f247b 2018-12-30)
* **IDE name and version:** CLion 2018.3.3
* **Operating system:** Windows 10 1809

## Problem description

A function taking another function as callback where parameter names match will fail the plugin's check for duplicate names, but pass rustc's.

## Steps to reproduce

```rust
fn foo(a: i32, b: fn (a: i32)) {}
```

This reports an error in the IDE but not to `cargo check`.
